test/tests.h

```diff
@@ -45,9 +46,9 @@ size_t encode(char *dest, const char *buf)
                dest[d] = 0; /* NUL-terminate destination string */
                return i + 1;
           }
-          check(sscanf(buf + i, "%x", &c) == 1, "invalid hex input %s", buf+i);
+          check(sscanf(buf + i, "%x", (unsigned int *)&c) == 1, "invalid hex input %s", buf+i);
           i = j; /* skip to char after hex input */
-          d += utf8proc_encode_char(c, (utf8proc_uint8_t *) (dest + d));
-     } while (1);
+          d += encode_char(c, (utf8proc_uint8_t *) (dest + d));
```

Comment:
It's actually, not putting `utf8proc_` prefix back, but rather adding `utf8proc_unsafe_`.
The version that starts with just `utf8proc_` would not work correctly...  My jetlagged brain missed that test case :sad: